server/item/recipe/recipe.go

```diff
@@ -0,0 +1,43 @@
+package recipe
+
+import "github.com/df-mc/dragonfly/server/item"
+
+// Recipe is implemented by all recipe types.
+type Recipe interface {
+	// Inputs returns the items required to craft the recipe.
+	Inputs() []InputItem
```

Comment:
Would maybe name this `Input` to match the naming of `Output`. Of course Output only returns one `item.Stack`, but `Input` is a valid plural too so that might be the way to go.